src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveryCompleteEventArgs.cs

```diff
@@ -53,30 +84,42 @@ public DiscoveryCompleteEventArgs(long totalTests, bool isAborted)
     /// <summary>
     ///   Indicates the total tests which got discovered in this request.
     /// </summary>
+    [DataMember]
     public long TotalCount { get; private set; }
 
     /// <summary>
     /// Specifies if discovery has been aborted. If true TotalCount is also set to -1.
     /// </summary>
+    [DataMember]
     public bool IsAborted { get; private set; }
 
     /// <summary>
     /// Metrics
     /// </summary>
+    [DataMember]
     public IDictionary<string, object> Metrics { get; set; }
 
     /// <summary>
     /// Gets the list of sources which were fully discovered.
     /// </summary>
+    [DataMember]
     public IList<string> FullyDiscoveredSources { get; set; }
 
     /// <summary>
     /// Gets the list of sources which were partially discovered (started discover tests, but then discovery aborted).
     /// </summary>
+    [DataMember]
     public IList<string> PartiallyDiscoveredSources { get; set; }
 
     /// <summary>
     /// Gets the list of sources which were not discovered at all.
     /// </summary>
+    [DataMember]
     public IList<string> NotDiscoveredSources { get; set; }
+
+    /// <summary>
+    /// Gets or sets the collection of discovered extensions.
+    /// </summary>
+    [DataMember]
+    public IDictionary<string, ISet<string>> DiscoveredExtensions { get; set; }
```

Comment:
Avoid null by doing init to empty dictionary?

Reply:
Done.